tests/unit/linalg/test_hypre_vector.cpp

```diff
@@ -65,6 +78,101 @@ TEST_CASE("HypreParVector I/O", "[Parallel], [HypreParVector]")
    remove((prefix + "INFO." + suffix).c_str());
 }
 
+TEST_CASE("HypreParVector Move Constructor", "[Parallel], [HypreParVector]")
+{
+   int world_size, rank;
+   MPI_Comm_size(MPI_COMM_WORLD, &world_size);
+   MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+   int size_per_rank = 2;
+
+   HYPRE_BigInt glob_size = world_size*size_per_rank;
+   std::vector<HYPRE_BigInt> col = detail::GenerateColumnPartitioning(world_size,
+                                                                      rank, size_per_rank);
+
+   // Initialize vector
+   HypreParVector v1(MPI_COMM_WORLD, glob_size, col.data());
+   for (int i=0; i<size_per_rank; ++i)
+   {
+      v1(i) = rank*size_per_rank + i;
+   }
+
+   // Make a copy so we can compare it later
+   HypreParVector v1_copy(v1);
+   Vector v1_base_copy(v1);
```

Comment:
I don't think you need these "`base`" vectors, you can index directly into the `HypreParVector`.